paddle/phi/kernels/onednn/interpolate_kernel.cc

```diff
@@ -15,11 +15,41 @@
 #include "paddle/phi/kernels/interpolate_kernel.h"
 
 #include "paddle/phi/backends/onednn/onednn_reuse.h"
+#include "paddle/phi/core/compat/get_kerneltype_forvar_utils.h"
 #include "paddle/phi/core/kernel_registry.h"
 #include "paddle/phi/kernels/funcs/interpolate_function.h"
 
 namespace phi {
 
+KernelKey InterpolateGetKernelTypeForVar(
+    const GetKernelTypeForVarContext* ctx) {
+  const std::string& var_name = ctx->GetVarName();
+  const DenseTensor& tensor = ctx->GetTensor();
+  const KernelKey& expected_kernel_type = ctx->GetKernelKey();
+  const AttributeMap& attrs = ctx->GetAttrs();
+  // Only input require reshaping, weights and
+  // bias are having shape in NCHW order
```

Comment:
I'm not sure if I understand these correctly. Do you mean "Only input requires changing data_layout"?
Usually, we distinguish shape and layout. Take `Tensor{shape={1,3,8,8}, data_format={"NCHW"},stride={192,64,8,1}}` as an example. `data_format` means the shape is ordered by NCHW(a.k.a channel_size is 3). stride means it has contiguous data_layout(DataLayout::kNCHW). We can reorder the tensor to kNHWC/KONEDNN but keep the shape unchanged. We can reshape the tensor to {3,8,8}/{24,8}/{1,192} with the layout unchanged.

Reply:
I think that `shape` `data_format`and`stride` is  a whole, together to represent a tensor。Take `Tensor{shape={1,3,8,8}, data_format={"NCHW"},stride={192,64,8,1}} `as an example. If you change the `data_format` from `NCHW` to `NHWC`, the `shape` must change from `{1,3,8,8}` to `{1, 8, 8, 3}` together. Otherwise, the data is wrong when you use.